src/probestack_x86.S

```diff
@@ -0,0 +1,38 @@
+// This is the same as x86_64, only translated for 32-bit sizes. Note that on
+// Unix we're expected to restore everything as it was, this function basically
+// can't tamper with anything.
+//
+// The ABI here is the same as x86_64, except everything is 32-bits large.
+
+.text
+.globl  __rust_probestack
+.type   __rust_probestack, @function
+__rust_probestack:
+    .cfi_startproc
```

Comment:
To confirm, are we *certain* that this can't be done with LLVM's inline assembly? For example does updating ebp/rbp work for gdb/backtraces? Does LLVM inline assembly not support the `.cfi_*` directives?

(in general it'll just be easier if we can stick to inline assembly I think)

Reply:
I've tried adding the CFI directives both in the original inline assembly, also via `global_asm!`, and both cases they did not result in a binary containing the directives, as observed via `dwarfdump`. Only a separate compilation worked. I haven't tested yet whether `gdb` picks on the `ebp`/`rbp` usage without the directives.

Reply:
gdb is very picky about what kind of prelude it accepts. Even placing a `rex` prefix before the `push` breaks it: https://github.com/bjorn3/rustc_codegen_cranelift/issues/146#issuecomment-449474527. Because of this, I wouldnt be suprised if using a different register as base pointer confuses gdb.

Reply:
I think we can use the standard base register prelude here, and if that works I think we probably want to stick with that since we should be able to continue using inline assembly. 

@da-x want to test on your end with the base pointer management added but still using inline assembly and seeing if it works?

Reply:
It works as expected. Please consider #306 and close this PR.